lib/matplotlib/legend_handler.py

```diff
@@ -776,7 +776,7 @@ class HandlerPolyCollection(HandlerBase):
     """
     def _update_prop(self, legend_handle, orig_handle):
         def first_color(colors):
-            if colors.size == 0:
+            if np.array(colors).size == 0:
```

Comment:
```suggestion
            if np.asarray(colors).size == 0:
```
is probably slightly better (as it will avoid copying data for all the artists that are well behaved). Also, if we now convert it to an array, why not use that below? An option is probably to use `len` instead. Or make sure that the problematic artist returns a NumPy array.

Reply:
Looking a bit more into it, I think a better fix is to do `return np.asarray(self._facecolors2d)` here:
https://github.com/matplotlib/matplotlib/blob/a9afe5b8ab420c4bf65a38f2d78bb4badc88a5b8/lib/mpl_toolkits/mplot3d/art3d.py#L1066
and correspondingly here:
https://github.com/matplotlib/matplotlib/blob/a9afe5b8ab420c4bf65a38f2d78bb4badc88a5b8/lib/mpl_toolkits/mplot3d/art3d.py#L1074

(or make them NumPy-arrays here:
https://github.com/matplotlib/matplotlib/blob/a9afe5b8ab420c4bf65a38f2d78bb4badc88a5b8/lib/mpl_toolkits/mplot3d/art3d.py#L1005-L1006
which seems to be the only code path where they are not NumPy-arrays.)

Reply:
Thanks, @oscargus! I was initially hesitant to touch anything else to avoid conflicts. But your suggestions work perfectly fine!